### DISCUSSION TITLE
Digitizer - AI on the edge - watermeter || Fehlablesung #786

### DISCUSSION THREAD

**Discussion by Takahara2022** [jomjol/AI-on-the-edge-device#2271] [Q&A]
Hallo, 

ich beziehe mich auf einen alten Post #786 und habe Probleme bei der Ablesung der digitalen ROI's beim Übergang bzw. Wechsel der ROI's. Mein neuer Wasserzähler zeigt die digitalen ROI's beim Übergang nicht vollständig an und daher kann die Software die digitalen Werte nicht richtig ermitteln.
Ich habe schon versucht die ROI's neu einzustellen (Größe und Position), kann aber für meinen Wasserzähler nicht das passende Setup ermitteln. Hat jemand Ideen was ich noch ändern kann um auch die Ablesung beim Übergang bzw. Wechsel der ROI's zu ermöglichen?

Vielen Dank im Vorraus.
Takahara
![2023-03-31 06_09_59-Window](https://user-images.githubusercontent.com/104391155/229021827-96cf9302-dc6f-418b-8265-a536837bc3f2.png)
![2023-03-31 06_19_27-Window](https://user-images.githubusercontent.com/104391155/229021831-afbaa90c-96eb-4f2c-b918-c5c825795223.png)

**Comment by ARPIpappen** [comment]
Ist bei mir aktuell ähnlich. Bin gespannt, welche Lösungen es gibt.

**Comment by wg25** [comment]
Welches Model für die digitale Analyse verwendet ihr? Sieht nach einem dig-class aus, versuch mal dig-cont.

Bei mir:
[Digits]
Model = /config/dig-cont_0611_s3.tflite

**Reply by Takahara2022** [reply]
Moin,
ich hab zwar kein Plan von was du da sprichts, aber ich einfach mal in die Config.ini Datei geschaut und folgende gefunden:

[Digits]
Model = /config/dig-s2-q-20220211.tflite

**Reply by ghost** [reply]
Die erste Version, die heute dig-class11 heißt hatte @wg25 gemeint. dig-s2-20220211 ist so eines.

Aktuellere Modelle wie das dig-cont_0611 oder [dig-class100-0161_s2_q.tflite](https://github.com/haverland/Tenth-of-step-of-a-meter-digit/blob/master/output/dig-class100-0161_s2_q.tflite) sind auf mehr Bildern trainiert und in der Erkennung genauer. Außerdem hat man das Problem mit den "N"s nicht, da diese Modelle auch Zwischenstände beim Umschalten der Ziffer erkennen.

**Reply by Takahara2022** [reply]
Vielen Dank für deine Antwort. Wie kann ich an diese neuere Version kommen? 
Kann ein Update auf Release 15.1.1 das Problem lösen? Ich nutze aktuell noch die HTML Version 13.1.1.

**Reply by ghost** [reply]
In den neuen Versionen sind natürlich auch neue Modelle mit dabei. Wobei selbst bei der 13.x schon neuere Modelle bei sind. Die wählst du einfach in der Configuration aus.

Beim Update würde ich keine Hauptversion überspringen. Das könnte zu Problemen führen. HTML wird automatisch mit der Update.zip mit ausgeteilt, so dass du dir da keine Sorgen machen musst.

**Reply by ghost** [reply]
Achse und die Modelle einfach via Fileupload in das config - Verzeichnis legen. Danach kann man sie in der Configuration auswählen.

**Comment by Takahara2022** [comment]
Ich versuche die Version 15.1.1 zu installieren und habe den ESP32 erfolgreich über das WEB Tool geflasht und die die Dateien auf die SD-Card geladen. Leider Ohne Erfolg :(

Fehlermeldung 1: In the selectd field the value '/config/ana-class100_0157_s1_q.tflite' in the section 'Analog' in the field 'Model' is invalid.
PLEASE CHECK BEFORE SAVING!

Fehlermeldung 2: In the selectd field the value '/config/dig-cont_0611_s3.tflite' in the section 'Digits' in the field 'Model' is invalid. PLEASE CHECK BEFORE SAVING!

Im Verzeichnung \config sind allerdings die Dateien "ana-class100_0157_s1_q.tflite" und "dig-cont_0611_s3_q.tflite". 

![Config](https://user-images.githubusercontent.com/104391155/229357213-a2a4d9b0-b476-4458-a1fa-96e5d760cb89.png)

**Reply by ghost** [reply]
Kannst du eines der Modelle in der Config auswählen? Am Besten mal einen Screenshot von der Auswahlbox machen.

**Reply by Takahara2022** [reply]
Ich kann keine Modelle in der Config auswählen. 
![Config](https://user-images.githubusercontent.com/104391155/229364087-a544657a-f63b-4a75-99fa-c6095b71fd71.png)

**Reply by Slider0007** [reply]
Deine Oberfläche schaut aber nicht nach v15.1.1 aus.

Prüfe bitte unter System->Info angezeigte Version für Webinterface and Firmware.

**Reply by Takahara2022** [reply]
Die System-> Info Seite sieht bei mir aktuell wie folgt aus:
![System_Info](https://user-images.githubusercontent.com/104391155/229370588-df28b18b-74f8-4906-b49e-16e25ae35107.png)

Die Overview Seite sieht so aus:

![Overview](https://user-images.githubusercontent.com/104391155/229370594-9866843b-8499-4015-8f65-b69a3c53a6f9.png)

**Reply by ghost** [reply]
Ich vermute, dass bei dir das HTML-Update nicht funktioniert hat. Seit der Version 13.x sollte auf der Hauptseite unten links ganz klein die Version stehen:

<img width="192" alt="image" src="https://user-images.githubusercontent.com/412645/229375929-489bd959-52e9-42df-b312-e24fc9874b91.png">

Aber das Menü sieht bei der Version 14 und 15 so aus:
<img width="569" alt="image" src="https://user-images.githubusercontent.com/412645/229376013-3ee4386c-d333-4a71-8fd1-341e578b9f78.png">

Wenn bei dir die Firmware aktualisiert wurde, aber nicht das Verzeichnis html, kann es gut sein, dass jetzt einiges nicht geht. Am besten aus dem update.zip den HTML-Ordner auf die SD-Karte kopieren.

**Comment by Takahara2022** [comment]
Vielen Dank @haverland  für deine Hilfe. Nach ich noch mal den HTML-Ordner auf die SD-Karte kopiert hatte, scheint es jetzt zu funktionieren.
![Wasserzähler](https://user-images.githubusercontent.com/104391155/229976136-c59c3548-1812-43c0-b697-7d0f00a6fd39.png)
